### DISCUSSION TITLE
Make the experience less painful for running Otterscan locally

### DISCUSSION THREAD

**Discussion by wmitsuda** [otterscan/otterscan#20] [Ideas]
Currently, it requires a lot of setup to run Otterscan. The scope of this discussion is only for running Otterscan, not about patching Erigon.

The current process described in README was based on my personal experience running it as a Otterscan developer. But for someone who just wants to run it, it can be painful.

The first idea I think we could explore is to publish docker images with the build result, find some way to also publish the 4bytes and trustwallet assets in a way that they can be easily downloaded, I have yet to test if docker storage images are a viable solution.

And lastly, they need all to be run together.

I'm open to hear other viable alternatives.

**Comment by androolloyd** [comment]
i have created an entire docker-compose package for dappnode that boots erigon, otterscan and abstracts out the services, i nixed the trustwallet repo and just connect put the raw github to it as an ENV variable now.

**Reply by androolloyd** [reply]
This was just updated to match the changes in #19 https://github.com/dappnode/DAppNodePackage-Erigon

**Reply by wmitsuda** [reply]
@androolloyd I want to apologized in advance, I was playing with docker builds today I hope of simplifying how people could run production builds, and I ended up build a unique docker image containing otterscan + 4bytes + trustwallet and serving them from nginx (the serve npm package was crashing sometimes with 4bytes, so I intend to use nginx from now on).

The bad (?) news is that it makes your patch in the other PR unnecessary, because from now on, everything is deployed together. Could you please try out the `feature/docker-experiments` branch and tell me what do you think about it?

I'm still thinking of some solution for overwriting certain parameters at runtime, unfortunately React generates a static website, so solutions like env variables only work at build time and requires a complete rebuild every time they change.

I'm almost reaching a solution involving some script magic for customizing the erigon URL at runtime, it is the only thing blocking me from promoting this new format to be the official one.

**Reply by wmitsuda** [reply]
I updated the README also, with the new instructions, please let me know if I forgot something important.
